bin/cqlsh.py

```diff
@@ -481,12 +484,18 @@ def __init__(self, hostname, port, color=False,
                                                        load_balancing_policy=WhiteListRoundRobinPolicy([self.hostname]))
             }
 
-            self.conn = Cluster(contact_points=(self.hostname,), port=self.port, cql_version=cqlver,
+            if cloudconf is None:
+                kwargs['contact_points'] = (self.hostname,)
+                kwargs['port'] = self.port
+                kwargs['ssl_options'] = sslhandling.ssl_settings(hostname, CONFIG_FILE) if ssl else None
+                profiles[EXEC_PROFILE_DEFAULT].load_balancing_policy = WhiteListRoundRobinPolicy([self.hostname])
```

Comment:
The code in this repo is up to date with cassandra trunk, so it a bit differently from the 3.11 branch code you are working with in tools-java

That's fine.

Reply:
Ah, I see it uses execution profiles now. Still, the logic is weird:
```

            profiles = {
                EXEC_PROFILE_DEFAULT: ExecutionProfile(consistency_level=cassandra.ConsistencyLevel.ONE,
                                                       request_timeout=request_timeout,
                                                       row_factory=ordered_dict_factory,
                                                       load_balancing_policy=WhiteListRoundRobinPolicy([self.hostname]))
            }

            if cloudconf is None:
                kwargs['contact_points'] = (self.hostname,)
                kwargs['port'] = self.port
                kwargs['ssl_options'] = sslhandling.ssl_settings(hostname, CONFIG_FILE) if ssl else None
                profiles[EXEC_PROFILE_DEFAULT].load_balancing_policy = WhiteListRoundRobinPolicy([self.hostname])
```
So, load_balancing_policy is always `WhiteListRoundRobinPolicy([self.hostname]))` - even if cloudconf is set - as that's the value used when creating profiles variable.

Did you check if it works when using cloudconf? I suspect it may not if the policy is really `WhiteListRoundRobinPolicy([self.hostname]))`

Reply:
you are correct, I need to remove it.

we can't limit to one node with the cloud bundle and `WhiteListRoundRobinPolicy`.
I did it for SCT code with the `HostFilterPolicy`, but it's need prior knowledge of that one node host_id.

https://github.com/scylladb/scylla-cluster-tests/pull/5372/files#diff-31591c8652c0f342cb95b98a23dd0f8886160c29b7028b7f4807225158041cc6R3348

Reply:
fixed it